lib/objects/objectsInMemFileDB.js

```diff
@@ -2569,6 +2574,52 @@ class ObjectsInMemoryFileDB extends InMemoryFileDB {
             this.writeTimer = null;
         }
     }
+
+    isEquivalentValue(a, b, ignoreField) {
+        // Create arrays of property names
+        if (a === null || a === undefined || b === null || b === undefined) {
+            return (a === b);
+        }
+        const aProps = Object.getOwnPropertyNames(a);
+        const bProps = Object.getOwnPropertyNames(b);
+
+        const lengthModA = a[ignoreField] !== undefined ? 1 : 0;
+        const lengthModB = b[ignoreField] !== undefined ? 1 : 0;
+
+        // If number of properties is different,
+        // objects are not equivalent
+        if ((aProps.length - lengthModA) !== (bProps.length - lengthModB)) {
+            //console.log('num props different: ' + JSON.stringify(aProps) + ' / ' + JSON.stringify(bProps));
+            return false;
+        }
+
+        for (let i = 0; i < aProps.length; i++) {
+            const propName = aProps[i];
+            if (propName === ignoreField) continue;
+
+            if (typeof a[propName] !== typeof b[propName]) {
+                //console.log('type props ' + propName + ' different');
+                return false;
+            }
+            if (typeof a[propName] === 'object') {
```

Comment:
that is true when `a[propName] === null`, but I guess you're checking that in the next recursion.

Reply:
Hmm, this is also true when `a[propName]` is an array and in that case `getOwnPropertyNames` includes the `length` field which is not enumerable. Might be confusing.

You might want to use these methods to check:
https://github.com/ioBroker/ioBroker.js-controller/blob/master/lib/tools.js#L1834-L1851

Reply:
> On a side note, the two eslint rules conflict with BF's code style. @foxriver76
These rules were introduced, based on a formatting commit BF has done, so I guess BF isn't sure about which code style to use ;-)

Reply:
Whoops :D

Reply:
Yes you are right array can also be compared directly ... or?!

Reply:
no, we can't

Reply:
interestingly it works :-) it just checks one more  entry which is length